src/components/TextField/TextField.test.tsx

```diff
@@ -16,6 +16,19 @@ describe('<TextField />', () => {
     it('matches its snapshot', () => {
       expect(component).toMatchSnapshot();
     });
+
+    it('focus', () => {
+      const instance = component.instance() as any;
+      const fabricTextFieldElement = { focus: jest.fn(), setSelectionRange: jest.fn() };
+      const focusSpy = jest.spyOn(fabricTextFieldElement, 'focus');
+      const setSelectionRangeSpy = jest.spyOn(fabricTextFieldElement, 'setSelectionRange');
+
+      instance.fabricTextFieldElement = fabricTextFieldElement;
```

Comment:
It started to feel a bit dirtier, so I backed out. It would need to be in a separate file, since the snapshot tests depend on the real fabric text field, then I felt a bit challenged about making a fake ref function that would return fake ref.

Reply:
I took another run at it. This time using a full mount and spying on functions on the prototype of the fabric component. I think this should address your concerns about brittleness. 

I spent some time trying `jest.mock('office-ui-fabric-react/lib/TextField', ...)` but failed to find a way to construct a fake react component that took props. (Since the YamUI Textfield depends on Fabrics componentRef, the fake would need to replicate it's behavior. )